src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -81,6 +84,7 @@ export async function prepareDocument(
                 highlightDocument(registry, uri)
             }
         })
+        clearStreamIdMapOnDocumentClose(registry, doc)
```

Comment:
btw, this and the `onDidChangeTextDocument` result above it, should probably be stored with the registration in the `registry`. And then when the `registry` entry is destroyed, these handlers should be disposed.

This is the risk with "registries" of all kinds: have to manage the lifecycle. If we don't dispose these things, we will accumulate memory leaks.

Reply:
onDidCloseTextDocument() should be called exactly once for the lifetime of the Toolkit, rather than once per prepareDocument() call.

Then just add it to subscriptions.push(...).

Reply:
I am assuming the second comment here is out of scope of this PR? Should I make a ticket for this?

Reply:
I'd do it in a separate PR IMO